<launch>
    <!-- my_navigation -->
    <node name="my_navigation" pkg="base_move" type="my_navigation_node" output="screen">
        <param name="active" value="true"/>

        <param name="move_frequency" value="60"/>
        <param name="speed_frequency" value="100"/>
        <param name="init_pose_x" value="0"/>
        <param name="init_pose_y" value="0"/>
        <param name="init_pose_yaw" value="$(eval 0 * 3.1415 / 180.0)"/>
        <param name="MAX_linear_speed" value="0.5"/>        <!-- [m/s] -->
        <param name="MAX_angular_speed" value="0.8"/>       <!-- [rad/s] -->
        <param name="linear_acceleration" value="0.05"/>    <!-- [m/s^2] -->
        <param name="linear_deceleration" value="0.05"/>    <!-- [m/s^2] -->
        <param name="angular_acceleration" value="0.2"/>    <!-- [rad/s^2] -->
        <param name="angular_deceleration" value="0.2"/>    <!-- [rad/s^2] -->
        <param name="linear_margin" value="0.001"/>         <!-- [m] -->
        <param name="angular_margin" value="0.001"/>        <!-- [rad] -->
        <param name="stop_margin" value="0.0001"/>          <!-- [m/s & rad/s] -->
  
        <param name="twist_topic" value="/cmd_vel"/>            <!-- Pub [geometry_msgs::Twist] -->
        <param name="reached_topic" value="/reached_status"/>   <!-- Pub [std_msgs::Bool] -->
        <param name="mapPose_topic" value="/map_pose"/>         <!-- Sub [geometry_msgs::PoseStamped] -->
        <param name="odom_topic" value="/odom"/>                <!-- Sub [nav_msgs::Odometry] -->
        <param name="goal_topic" value="/base_goal"/>           <!-- Sub [geometry_msgs::Pose] -->
    </node>

    <!-- odometry -->
    <include file="$(find odometry)/launch/odometry.launch"/>
    
    <!-- fake_odom -->
    <!-- <include file="$(find fake_odom)/launch/fake_odom.launch"/> -->

    <!-- tf2_transformListener -->
    <node name="tf2_transformListener" pkg="base_move" type="tf2_transformListener" output="screen"/>

    <!-- main program -->
    <node name="main_node" pkg="main_program" type="main_tel2022" output="screen"/>
</launch>